docs/SpellExample.scala

```diff
@@ -0,0 +1,277 @@
+
+    val addDragonResults = await(client.query(
+      Create(Class(Value(SPELLS_CLASS)),
+        Obj("data" ->
+          Obj("name" -> "Water Dragon's Claw", "element" -> "water")
+        )
+      )
+    ))
+    println(s"Added spell to class $SPELLS_CLASS \n $addDragonResults \n")
+
+    val addHippoResults = await(client.query(
+      Create(Class(Value(SPELLS_CLASS)),
+        Obj("data" ->
+          Obj("name" -> "Hippo's Wallow", "element" -> "water", "cost" -> 35)))
+    ))
+    println(s"Added spell to class $SPELLS_CLASS:\n $addHippoResults \n")
+
+    //The results at 'ref' are a resource pointer to the class that was just created.
+    val hippoRef = addHippoResults("ref")
+    println(s"hippoRef = $hippoRef \n")
+
+    /*
+    * Read the hippo back that we just created
+    */
+    val getHippoResults: Value = await(client.query(
+      Select(Value("data"), Get(hippoRef))
+    ))
+    println(s"Hippo Spell:\n $getHippoResults \n")
+
+    //convert the hippo results into primitive elements
+    val name: String = getHippoResults("name").to[String].get
+    val cost: Int = getHippoResults("cost").to[Int].get
+    val element: String = getHippoResults("element").to[String].get
+    println(s"Spell Details: Name=$name, Cost=$cost, Element=$element")
+
+    //This would return an empty option if the field is not found or if the conversion fails
+    val optSpellElement: Option[String] = getHippoResults("element").to[String].toOpt
+    optSpellElement match {
+      case Some(element2) => println(s"optional spell element $element2")
+      case None => println("Something went wrong reading the spell")
+    }
+
+    /*
+     * Query for all the spells in the index
+     */
+    val queryIndexResults: Value = await(client.query(
+      SelectAll(Path("data", "id"),
+        Paginate(
+          Match(Index(Value(INDEX_NAME)))
+        )
+      )
+    ))
+
+    val spellsRefIds: Seq[String] = queryIndexResults.to[Seq[String]].getOrElse(Seq.empty)
+    println(s"spellsRefIds = $spellsRefIds \n")
+
+    /*
+    * Store a Spell case class
+    */
+    val newSpell = Spell("Water Dragon's Claw", "water", Option(25))
+    val storeSpellResult = await(client.query(
+      Create(
+        Class(SPELLS_CLASS),
+        Obj("data" -> newSpell))
+    ))
+    println(s"Stored spell:\n $storeSpellResult \n")
+
+    /*
+     * Read the spell we just created
+     */
+    val dragonRef = storeSpellResult("ref")
+    val getDragonResult = await(client.query(
+      Select(
+        Value("data"),
+        Get(dragonRef)
+      )
+    ))
+
+    val spell = getDragonResult.to[Spell].get
+    println(s"dragon spell: $spell \n")
+
+    /*
+    * Example of using proper Future mapping to handle results async.  Could be map, flatMap or a for expression
+    */
+    for {
+      dragonSpellVal <- client.query(Select(Value("data"), Get(dragonRef)))
+      dragonSpell = dragonSpellVal.to[Spell].get
```

Comment:
not really sure, but can't you do `dragonSpell <- dragonSpellVal.to[Spell]`?

Reply:
Without the `get` what I see is it returns `Result[Spell]`.  So I think the `get` is necessary to return the actual Spell?